pkg/models/datasource_cache.go

```diff
@@ -127,14 +127,13 @@ func (ds *DataSource) HTTPClientOptions() (*sdkhttpclient.Options, error) {
 		}
 	}
 
-	if ds.JsonData != nil && ds.JsonData.Get("azureAuth").MustBool() {
+	if ds.JsonData != nil {
```

Comment:
Thanks for the feedback, @marefr

I saw that the `datasource_cache` used only by plugin proxy and Prometheus plugin is transitioning to implementation in Plugin SDK (e.g. [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/4c305c4d10215404291b5c2963a83f7acd558264/backend/http_settings.go#L128)) and things like "test connection" data dashboards already use that implementation.

It needs to be implemented in Plugin SDK and I will send a separated PR for that. And we would discuss that.

I would want to merge this PR before because **this PR doesn't introduce any new design concepts or assumptions** but it cleans up the schema of Azure credentials in JsonData and removes unnecessary knowledge of Azure credentials schema from the `DatasourceHttpSettings` component.

This would be very helpful before further discussion on support of Azure credentials [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/4c305c4d10215404291b5c2963a83f7acd558264/backend/http_settings.go#L128) etc.

Right now this is under experimental feature flag `httpclientprovider_azure_auth` in Plugin SDK.


#### `azureCredentials` vs `_azureCredentials`

The original JSON object that comes from UI is `azureCredentials`. But that object is a raw set of key/values which are not validated. `_azureCredentials` holds  `AzureCredentials`, a domain object, abstract from how it's being serialized and stored by a plugin.

`datasource_cache` (as a builder for `httpclient.Options`), validates raw user input and deserializes `AzureCredentials` to `CustomOptions["_azureCredentials"]`.

So, `azureCredentials` is RAW key/value pairs received from frontend, `_azureCredentials` is a parsed `AzureCredentials` object.

Azure middleware (and Azure Token Provider) works with `AzureCredentials`.

For example, Azure Monitor has own, already established, JSON schema for Azure credentials and it has own implementation of parsing it from JsonData settings. Having `AzureCredentials` in `httpclient.Options`  allows to abstract out of variations in JsonData schemas of different plugins, and reuse the same generic Azure middleware implemented in `httpclient`/`httpclientprovider`.

For example, Azure Monitor could just provide credentials [here](https://github.com/grafana/grafana/blob/96efbbaed1728e89602ca9578a24d24bd2ceb53c/pkg/tsdb/azuremonitor/httpclient.go#L42-L41) to `httpclient.Options` instead of implementing and passing own middleware:

```go
func newHTTPClient(route azRoute, model datasourceInfo, cfg *setting.Cfg, clientProvider httpclient.Provider) (*http.Client, error) {
	m, err := getMiddlewares(route, model, cfg)
	if err != nil {
		return nil, err
	}

	return clientProvider.New(httpclient.Options{
                AzureCredentials: model.Credentials, // CustomOptions["_azureCredentials"]
		Headers:          route.Headers,
		Middlewares:      m,
	})
}
```

Reply:
Hey @kostrse. Sorry for the somewhat late response. 

We've had some internal discussions and there's a need that have arise  for enable azure authentication for azure data sources not builtin to Grafana, e.g. external ones. Historically when we discussed this everything was experimental, but now feels like a more appropriate time for considering adding things to the grafana-plugin-sdk-go repository.

> I would want to merge this PR before because **this PR doesn't introduce any new design concepts or assumptions** but it cleans up the schema of Azure credentials in JsonData and removes unnecessary knowledge of Azure credentials schema from the `DatasourceHttpSettings` component.
> 
> This would be very helpful before further discussion on support of Azure credentials [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/4c305c4d10215404291b5c2963a83f7acd558264/backend/http_settings.go#L128) etc.

Makes sense. And it seems like you're already planning to go in the direction I'm suggesting below after this has been merged :+1: I'll go ahead and review this again so we can get it merged. I think I have your email since earlier meeting so I'll just send an email in case we need to sync up on things.

> It needs to be implemented in Plugin SDK and I will send a separated PR for that. And we would discuss that.

Great. We would like the SDK to include the settings/properties for azure authentication similar to the AWS SigV4 settings as can be seen [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/329c72e4e242f84c3b185013f4863fcce7996086/backend/http_settings.go#L39-L46), but keep the SDK as agnostic as possible. With that said, it would be good to keep the shared logic for azure authentication in a separate repository so that Grafana and external plugins could make use of it. Seems like there's an existing repository created, [grafana-azure-sdk-go](https://github.com/grafana/grafana-azure-sdk-go), which seems like a good fit. We have this setup for AWS as inspiration/guidance, [grafana-aws-sdk](https://github.com/grafana/grafana-aws-sdk).

Reply:
Seems like CI refuses to start so you'll probably have to merge with latest changes from main to get it running.